### ISSUE TITLE
Support `@typeParam` in docs for TypeScript

### ISSUE THREAD

**Issue by marikaner** [gajus/eslint-plugin-jsdoc#844]
## Motivation

We have a TypeScript project, where generic type parameters are often forgotten. 
Example of the context:
```ts

/**
  * Class description.
  * @typeParam T - Generic type argument description.
  */
export class MyClass<T extends string> { }
```

## Current behavior

Can't find rules around this. I know that this is TSDoc rather than JSDoc, but given that TypeScript is supported and I am a huge fan of the rules you have so far, I thought it could make sense.

## Desired behavior

Similar to the rules around `@param`, I would imagine rules to check `@typeParam` for correctness, existence and syntax.

## Alternatives considered

I am not aware of tools that would fulfil this, therefore the only alternative I see would be to build it on my own.

**Comment by brettz9** [comment]
You should be able to achieve this using the following:

1. For enforcing the presence of the `@typeParam` tag in a context, e.g., with `@template`: [`no-missing-syntax`](https://github.com/gajus/eslint-plugin-jsdoc#eslint-plugin-jsdoc-rules-no-missing-syntax).*
1. For enforcing presence of type and/or name with the `@typeParam` tag: [`settings.jsdoc.structuredTags`](https://github.com/gajus/eslint-plugin-jsdoc#eslint-plugin-jsdoc-settings-structuredtags)
1. For enforcing the presence or formatting of `@typeParam`'s description [`jsdoc/match-description`](https://github.com/gajus/eslint-plugin-jsdoc#match-description) with `tags` option

----
* Note that the specific AST we use here is our own and experimental; we recently discovered TypeScript has its own, so we might want to align the names at some point (though without losing the exact precision we support on whitespace) and/or I would hope get these standardized and supported by tools beyond TypeScript; I just mention this because it is possible any `contexts` you add for this rule targeting JSDoc such as JSDoc blocks with `@template` may need to be redone with different AST in the future.

**Comment by brettz9** [comment]
Closing as that should resolve the question.
